packages/workshop-app/app/components/toast.tsx

```diff
@@ -0,0 +1,216 @@
+				<ToastPrimitive.Root
+					asChild
+					forceMount
+					type="foreground"
+					duration={autoClose === false ? Infinity : duration}
+					open={true}
+					onOpenChange={e => {
+						setOpen(e)
+						onOpenChange?.(e)
+					}}
+				>
+					<motion.li
+						initial={{ y: 150, opacity: 0 }}
+						animate={{ y: 0, opacity: 1 }}
+						exit={{ y: 150, opacity: 0 }}
+						transition={{ ease: 'easeIn', duration: ANIMATION_DURATION / 1000 }}
+						className={clsx(
+							"grid grid-cols-[24px_1fr_16px] items-center gap-y-1 gap-x-2 rounded-md bg-white p-2 shadow-[hsl(206_22%_7%_/_35%)_0px_10px_38px_-10px,_hsl(206_22%_7%_/_20%)_0px_10px_20px_-15px] [grid-template-areas:_'icon_title_close'_'icon_description_description']",
+							{ 'border-l-8 border-red-500': variantColor === 'red' },
+							{ 'border-l-8 border-green-500': variantColor === 'green' },
+							{ 'border-l-8 border-blue-500': variantColor === 'blue' },
+						)}
+					>
+						{variant && (
+							<Icon
+								className={clsx(
+									'h-6 w-6',
+									{ 'text-red-500': variantColor === 'red' },
+									{ 'text-green-500': variantColor === 'green' },
+									{ 'text-blue-500': variantColor === 'blue' },
+								)}
+								viewBox="0 0 24 24"
+								name={variant}
+								aria-hidden="true"
+							/>
+						)}
+						<ToastPrimitive.Title className="font-bold [grid-area:_title]">
+							{title}
+						</ToastPrimitive.Title>
+						{children && (
+							<ToastPrimitive.Description
+								className="text-sm [grid-area:_description]"
+								asChild
+							>
+								{children}
+							</ToastPrimitive.Description>
+						)}
+						<ToastPrimitive.Close className="place-self-center rounded-full [grid-area:_close] hover:border-2 hover:border-current">
+							<Icon name="Close" aria-label="Close" />
+						</ToastPrimitive.Close>
+					</motion.li>
+				</ToastPrimitive.Root>
+			) : null}
+		</AnimatePresence>
+	)
+}
+
+// show empty line when the message contain two \n consecutively
+function splitMessageToRows(message: string | ToastProps['children'] = '') {
+	if (typeof message !== 'string') return message
+	const children = message
+		?.split('\n')
+		.map((line, index) =>
+			line ? (
+				<p key={index}>{line}</p>
+			) : (
+				<p key={index} className="block h-2"></p>
+			),
+		)
+	return <div>{children}</div>
+}
+
+interface ToastData extends ToastProps {
+	id: number
+	hash: string
+}
+
+const TOAST_EVENT_NAME: keyof CustomEventMap = 'kcdshop-toast-show'
+
+export function showToast(
+	element: EventTargetElement,
+	detail: ToastEventProps,
+) {
+	const event = new CustomEvent(TOAST_EVENT_NAME, { detail })
+	element?.dispatchEvent(event)
+}
+
+// display toast on client
+export function Toast(props: ToastEventProps) {
+	const [emitted, setEmitted] = useState(false)
+
+	useEffect(() => {
+		if (!emitted && typeof document === 'object') {
+			setEmitted(true)
+			showToast(document, props)
+		}
+	}, [emitted, props])
+
+	return null
```

Comment:
I think we can safely delete this as it's not used.

Reply:
Toast component it just an alternative way to show a toast, the other way is to use `showToast` function. you can use it to show notifications to the users.

I can remove it now, you can always bring it back in the future if you need it